src/Okta.Sdk/Api/OAuthApi.cs

```diff
@@ -183,10 +195,29 @@ public async Task<ApiResponse<OAuthTokenResponse>> GetBearerTokenWithHttpInfoAsy
             {
                 localVarRequestOptions.HeaderParameters.Add("Accept", localVarAccept);
             }
-
-            // make the HTTP request
+            
+            var dpopJwt = _dpopProofJwtGenerator.GenerateJWT();
+            localVarRequestOptions.HeaderParameters.Add("DPoP", dpopJwt);
+            
             var localVarResponse = await this.AsynchronousClient.PostAsync<OAuthTokenResponse>(accessTokenUri.Trim(), localVarRequestOptions, this.Configuration, cancellationToken).ConfigureAwait(false);
 
+            if (localVarResponse.StatusCode == HttpStatusCode.BadRequest 
+                && localVarResponse.Data.Error == "use_dpop_nonce"
+                && localVarResponse.Headers.TryGetValue("dpop-nonce", out var header))
+            {
+                var nonce = header.FirstOrDefault();
+                dpopJwt = _dpopProofJwtGenerator.GenerateJWT(nonce);
+                jwtSecurityToken = _jwtGenerator.GenerateSignedJWT();
+                
+                localVarRequestOptions.FormParameters.Remove("client_assertion");
+                localVarRequestOptions.FormParameters.Add("client_assertion", jwtSecurityToken.ToString());
+
+                localVarRequestOptions.HeaderParameters["DPoP"].Clear();
+                localVarRequestOptions.HeaderParameters["DPoP"].Add(dpopJwt);
+
+                localVarResponse = await this.AsynchronousClient.PostAsync<OAuthTokenResponse>(accessTokenUri.Trim(), localVarRequestOptions, this.Configuration, cancellationToken).ConfigureAwait(false);
+            }
```

Comment:
The new logic here should be delegated to the JwtGenerator implementation.

Reply:
See my comments below.